app/code/community/Metrilo/Analytics/Api/Connection.php

```diff
@@ -19,12 +19,10 @@ public function post($url, $bodyArray = false, $hmacAuth = false)
             'Connection: Close',
             'Host: '.$parsedUrl['host']
         ];
-    
-        if ($hmacAuth) {
-            $secret = $bodyArray['secret'];
-            unset($bodyArray['secret']);
-            $headers[] = 'X-Digest: ' . hash_hmac('sha256', json_encode($bodyArray), $secret);
-        }
+        
+        $secret = $bodyArray['secret'];
+        unset($bodyArray['secret']);
```

Comment:
I'm a bit puzzled why you get something from `$bodyArray` and unset it. It could be passed as a method argument. Also `$hmacAuth` flag is useless, but still present up there in the definition.

Reply:
$hmacAuth is a mistake left from the previous condition needed to add x-digest header only to monolith api calls. Will be removed. The secret is passed as argument for the helper responsible of the creation of the client object and passed down the way to the connection class. There it is needed only to create the hash_hmac hash, and will produce error if it is send with the rest of the parameters to Metrilo api. Adding that secret in this specific class will require additional load of resources that are already used in the client helper.